src/modules/geolocation/components/GeolocationInput.tsx

```diff
@@ -0,0 +1,105 @@
+  FormControl,
+  FormHelperText,
+  Stack,
+  Typography,
+} from '@mui/material';
+
+import React, { useId } from 'react';
+
+import { useGeolocation } from '../hooks/useGeolocation';
+import { ClearIcon, MyLocationIcon } from '@/components/elements/SemanticIcons';
+import { DynamicInputCommonProps } from '@/modules/form/types';
+import BaseMap from '@/modules/geolocation/components/BaseMap';
+import GeolocationErrorAlert from '@/modules/geolocation/components/GeolocationErrorAlert';
+import { LatLon } from '@/modules/geolocation/types';
+
+interface GeolocationInputProps
+  extends Pick<DynamicInputCommonProps, 'label' | 'helperText' | 'disabled'> {
+  value?: LatLon | null;
+  onChange: (value: LatLon | null) => void;
+}
+
+const GeolocationInput: React.FC<GeolocationInputProps> = ({
+  label,
+  value: coordinates,
+  onChange,
+  helperText,
+  disabled,
+}) => {
+  const { requestCoordinates, loading, error } = useGeolocation(onChange);
+  const labelId = useId();
+  const mapHeight = 230;
+
+  return (
+    <>
+      <FormControl sx={{ width: '100%' }}>
+        {label && <div id={labelId}>{label}</div>}
+        {error && <GeolocationErrorAlert error={error} />}
+        <Stack direction='row' sx={{ my: 0.5 }} gap={2} alignItems='center'>
+          <Button
+            onClick={requestCoordinates}
+            startIcon={<MyLocationIcon />}
+            variant='outlined'
+            aria-labelledby={label ? labelId : undefined}
+            disabled={disabled}
+          >
+            Request Location
+          </Button>
+          {coordinates && (
+            <Button
+              variant='gray' // TODO: update to outlined grayscale
+              onClick={() => onChange(null)}
+              startIcon={<ClearIcon />}
+            >
+              Clear Map
+            </Button>
+          )}
+        </Stack>
+
+        {helperText && <FormHelperText>{helperText}</FormHelperText>}
+        <Box sx={{ mt: 1 }}>
+          {!coordinates && !disabled && (
+            <Box
+              sx={{
+                display: 'flex',
+                justifyContent: 'center',
+                alignItems: 'center',
+                backgroundColor: 'background.default',
+                borderColor: 'borders.dark',
+                borderRadius: 1,
+                borderWidth: 1,
+                borderStyle: 'solid',
+                height: mapHeight,
+              }}
+              aria-live='polite' // Notify screen readers of content updates
+            >
+              {loading ? (
+                <Typography color='links' fontWeight={700}>
+                  Getting Location ...
+                </Typography>
+              ) : (
+                <Typography color='text.secondary'>
+                  Location not collected
+                </Typography>
+              )}
+            </Box>
+          )}
+          {coordinates && (
+            <Box
+              aria-live='polite' // Announce when the map is displayed
```

Comment:
I think this `aria-live='polite'` won't reliably announce, since aria-live announces that the component's internal text changes, but not when the component is added to or removed from the dom. (See "Live regions that are already "populated" don't work" https://tetralogical.com/blog/2024/05/01/why-are-my-live-regions-not-working/)

Locally I tried out moving the `aria-live='polite'` up to line 63 on the wrapping Box, but the issue with that is that it then announces all internal text changes to the contents, so when the map appears it announces "Leaflet | copyright OpenStreetMap contributors".

Reply:
Oh thanks for the reference. I tried to google this but came up with responses saying that it should work (as I interpreted, anyway) but I didn't test. So that is good to know

Reply:
This react explanation that was linked from the article is helpful. https://dev.to/abbeyperini/live-regions-in-react-4dmd
 I'll update this to the container is always rendered

Reply:
I got stuck here- moving along for now but I will come back to it

Reply:
https://github.com/open-path/Green-River/issues/7122